apt-hook/esm-templates.cc

```diff
@@ -0,0 +1,273 @@
+#include <apt-pkg/cachefile.h>
```

Comment:
I split `hook.cc` into three files:
* esm-templates.cc (this one) as most of the pre-existing code
* esm-templates.hh
* hook.cc is just a main function now

This was done so that we could use the `process_all_templates` function from the json-hook binary.